neuralprophet/plot_utils.py

```diff
@@ -538,3 +552,72 @@ def get_valid_configuration(  # move to utils
             "components_list": plot_components,
         }
     return valid_configuration
+
+
+def validate_current_env():
+    """
+    Validate the current environment to check if it is a valid environment to run the code.
+
+    Returns
+    -------
+    bool :
+        True if the current environment is a valid environment to run the code, False otherwise.
+
+    """
+    from IPython import get_ipython
+
+    if "google.colab" in str(get_ipython()):
+        log_warning_colab_resampler()
+        vaild_env = False
+    else:
+        if is_notebook():
+            vaild_env = True
+        else:
+            log_warning_static_env_resampler()
+            vaild_env = False
+    return vaild_env
+
+
+def is_notebook():
+    """
+    Determine if the code is being executed in a Jupyter notebook environment.
+
+    Returns
+    -------
+    bool :
+        True if the code is being executed in a Jupyter notebook, False otherwise.
+    """
+    try:
+        from IPython import get_ipython
+
+        if "IPKernelApp" not in get_ipython().config:  # pragma: no cover
+            return False
+    except ImportError:
+        return False
+    except AttributeError:
+        return False
+    return True
+
+
+def auto_set_plotting_backend(plotting_backend_original):
+    """Automatically set the plotting backend.
+
+    Given `plotting_backend_original`, returns "plotly-resample" if `validate_current_env()`
+    returns `True` and `plotting_backend_original` is "plotly-auto", "plotly" otherwise. If
+    `plotting_backend_original` is not "plotly-auto", returns `plotting_backend_original`.
+
+    Parameters
+    ----------
+    plotting_backend_original : str
+        Original plotting backend.
+
+    Returns
+    -------
+    str
+        The new plotting backend.
+    """
+    if plotting_backend_original == "plotly-auto":
+        plotting_backend_new = "plotly-resample" if validate_current_env() else "plotly"
+    else:
+        plotting_backend_new = plotting_backend_original
+    return plotting_backend_new
```

Comment:
I think you can use just `plotting_backend` as variable here - that will also shorten you code
```
if plotting_backend == "plotly-auto":
        plotting_backend = "plotly-resample" if validate_current_env() else "plotly"
    return plotting_backend
```

Further suggestion:
This only auto-sets the plotly backend.
Thus it could only be called if plotting backend is "plotly-auto" and not receive any input, and be called
`auto_set_plotly_backend()`